### PR TITLE
[fix](operation_node)Replace structure to ensure same colRef can be save 

### LINKED ISSUE #28438: [Bug] when use Minus or Except ，single node output column result incorrect, multiple nodes experience crash phenomenon
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

2.0.2

### What's Wrong?

multiple node 
image
single node 
image

### What You Expected?

fix it

### How to Reproduce?

Reproducing statements as below:

```sql
CREATE TABLE test_varchar(c1 varchar) DISTRIBUTED BY hash(c1) PROPERTIES ("replication_num" = "1");
CREATE TABLE test_date(c2 date) DISTRIBUTED BY hash(c2) PROPERTIES  ("replication_num" = "1");
insert into test_varchar values('+06-00');
insert into test_date values('1990-11-11');
SELECT c1, c1 FROM test_varchar MINUS SELECT c2, c2 FROM test_date;

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TomaYoko**
…be save

## Proposed changes

Issue Number: close #28438

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by TomaYoko** [comment]
run buildall

**Comment by TomaYoko** [comment]
run buildall

**Comment by TomaYoko** [comment]
run buildall

**Review by zhiqiang-hhhh — APPROVED** [review]
LGTM

File: regression-test/data/minus_except_test/minus_except.out [hunk 1]
```diff
@@ -0,0 +1,3 @@
+-- This file is automatically generated. You should know what you did if you want to edit this
+-- !select --
++06-00  +06-00
```
Comment (zhiqiang-hhhh) [inline regression-test/data/minus_except_test/minus_except.out:3]: incorrect content format
maybe:
`[+06-00, +06-00]`
or you can try
`sh run-regression-test --run -s minus_except -getOut`

**Comment by zhiqiang-hhhh** [comment]
@TomaYoko hello, thanks for your contribution, you can see my comment before buildall.

`sh run-regression-test --run -s minus_except -getOut` will make help

File: regression-test/suites/minus_except_test/minus_except.groovy [hunk 2]
```diff
@@ -24,7 +24,6 @@ suite("minus_except") {
     sql "CREATE TABLE ${tableNameDate}(c2 date) DUPLICATE key(c2) DISTRIBUTED BY hash(c2) BUCKETS auto PROPERTIES ('replication_num' = '1')"
     sql "insert into ${tableNameVarchar} values('+06-00');"
     sql "insert into ${tableNameDate} values('1990-11-11');"
-    qt_select " SELECT c1,c1 FROM ${tableNameVarchar} MINUS SELECT c2,c2 FROM ${tableNameDate};"
```
Comment (zhiqiang-hhhh) [inline regression-test/suites/minus_except_test/minus_except.groovy]: This test should be added back, change the result format PLS.

**Review by zhiqiang-hhhh — CHANGES_REQUESTED** [review]

File: regression-test/data/minus_except_test/minus_except.out (hunk 1 above)
Comment (TomaYoko) [reply regression-test/data/minus_except_test/minus_except.out:3]: This out file is generated by genOut command( I use the version of 2.0.2 )  and local test has passed , shall I change the version to master and regenerate the out file?
What I wonder is why the out file genterated  by local which use command below  can't passed by remote
command I used
image

Looking forward to your answer @zhiqiang-hhhh

File: regression-test/suites/minus_except_test/minus_except.groovy (hunk 2 above)
Comment (TomaYoko) [reply regression-test/suites/minus_except_test/minus_except.groovy]: > This test should be added back, change the result format PLS.

I added this test before,but it seems can't pass on the remote which reason may be due to grammar failure（the test can be tested successfully  locallly）
I wonder if is the same reason caused like  the one replied  above

File: regression-test/data/minus_except_test/minus_except.out (hunk 1 above)
Comment (zhiqiang-hhhh) [reply regression-test/data/minus_except_test/minus_except.out:3]: I tried on my local, gen out file content is same with yours

Comment (TomaYoko) [reply regression-test/data/minus_except_test/minus_except.out:3]: > I tried on my local, gen out file content is same with yours

so how can I get the result can be passed on remote 😂😂

**Comment by zhiqiang-hhhh** [comment]
@TomaYoko Leave the result file problem, I will fix it.
We still have some problem on PipelineX. 
maybe update `be/src/pipeline/pipeline_x/dependency.h` on line 499 in same way.
```cpp
//first:column_id, could point to origin column or cast column
//second:idx mapped to column types
std::unordered_map<int, int> build_col_idx;
```

BTW, PipelineX need branch master.

**Comment by zhiqiang-hhhh** [comment]
> > @TomaYoko Leave the result file problem, I will fix it. We still have some problem on PipelineX. maybe update `be/src/pipeline/pipeline_x/dependency.h` on line 499 in same way.
> > ```c++
> > //first:column_id, could point to origin column or cast column
> > //second:idx mapped to column types
> > std::unordered_map<int, int> build_col_idx;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > BTW, PipelineX need branch master.
> 
> @zhiqiang-hhhh Do I need to make the modification you mentioned together or keep it as it is? And whether the test I removed before should be added back( I wonder if this will cause a grammer failure again)

Just make it together and add the test case back.

**Comment by TomaYoko** [comment]
> > > @TomaYoko Leave the result file problem, I will fix it. We still have some problem on PipelineX. maybe update `be/src/pipeline/pipeline_x/dependency.h` on line 499 in same way.
> > > ```c++
> > > //first:column_id, could point to origin column or cast column
> > > //second:idx mapped to column types
> > > std::unordered_map<int, int> build_col_idx;
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > BTW, PipelineX need branch master.
> > 
> > 
> > @zhiqiang-hhhh Do I need to make the modification you mentioned together or keep it as it is? And whether the test I removed before should be added back( I wonder if this will cause a grammer failure again)
> 
> Just make it together and add the test case back.

@zhiqiang-hhhh OK,to look more neater, I've push a new request with your remind ,please help review it😀    https://github.com/apache/doris/issues/28438

**Comment by hello-stephen** [comment]
run p0

**Comment by hello-stephen** [comment]
run pipelinex_p0

**Comment by TomaYoko** [comment]
run P0

**Comment by TomaYoko** [comment]
run P0

**Comment by LemonLiTree** [comment]
run p0

**Comment by LemonLiTree** [comment]
run buildall

**Comment by TomaYoko** [comment]
run buildall

**Pull request closed without merging**